tests/testthat/test-with.R

```diff
@@ -39,3 +40,22 @@ test_that("can instantiate all linters without arguments", {
   expect_type(really_all_linters, "list")
   expect_length(really_all_linters, nrow(available_linters(exclude_tags = NULL)))
 })
+
+test_that("with_defaults is supported with a deprecation warning", {
+  defaults <- linters_with_defaults()
+  expect_warning(
+    old_defaults <- with_defaults(),
+    rex::rex("Use linters_with_defaults instead.")
+  )
+  expect_equal(defaults, old_defaults)
```

Comment:
identical?

Reply:
These are currently identical, but not e.g. if using `linters_with_tags("default")` as a reference, because this constructs new functions with their own environments.
That's why I decided to only test equality.